packages/embark/src/lib/modules/blockchain/index.js

```diff
@@ -22,47 +23,54 @@ class Blockchain {
       this.blockchainNodes[clientName] = startCb;
     });
 
-    this.events.setCommandHandler("blockchain:node:start", async (blockchainConfig, cb) => {
-      const self = this;
-      const clientName = blockchainConfig.client;
-      function started() {
-        self.startedClient = clientName;
-        self.events.emit("blockchain:started", clientName);
-      }
-      if (clientName === constants.blockchain.vm) {
-        started();
-        return cb();
-      }
-      const requestManager = new Web3RequestManager.Manager(blockchainConfig.endpoint);
-
-      const ogConsoleError = console.error;
-      // TODO remove this once we update to web3 2.0
-      // TODO in web3 1.0, it console.errors "connection not open on send()" even if we catch the error
-      console.error = (...args) => {
-        if (args[0].indexOf('connection not open on send()') > -1) {
-          return;
-        }
-        ogConsoleError(...args);
-      };
-      requestManager.send({method: 'eth_accounts'}, (err, _accounts) => {
-        console.error = ogConsoleError;
-        if (!err) {
-          // Node is already started
-          started();
-          return cb(null, true);
+    this.events.setCommandHandler("blockchain:node:start", async (initialBlockchainConfig, cb) => {
+      this.plugins.runActionsForEvent("blockchain:config:modify", initialBlockchainConfig, (err, blockchainConfig) => {
```

Comment:
Assuming this is being consumed using `registerActionForEvent("blockchain:config:modify")` in the ewasm plugin??

What does the plugin need to modify in the config that the user couldn't do directly in their Dapp's blockchain config?

Reply:
You can find it here: https://github.com/embark-framework/embark-nim-compiler/blob/master/index.js#L20-L35
Basically, we modify the config only if the user wants it (option set to `true`) and it,s just because there is a lot of configs that need to be set up to use Geth with the eWasm testnet, so the plugin offers the possibility to configure it for them.